packages/lesswrong/server/resolvers/commentResolvers.ts

```diff
@@ -105,4 +115,83 @@ augmentFieldsDict(Comments, {
       return (post && post.contents && post.contents.version) || "1.0.0";
     },
   },
+  doppelComments: {
+    resolveAs: {
+      type: '[DoppelComment]',
+      resolver: async (dbComment, args, context: ResolverContext) => {
+        const {DoppelComments, Users, currentUser} = context
+        if (!userHasDoppelComments(currentUser)) return []
+        // This should always be a single user
+        const [user] = await getWithLoader(context, Users, "usersByIdsForDoppelComments", {}, "_id", dbComment.userId, {karma: 1})
+        if (!user) return []
+        if (user.karma < 10_000) return []
+        const doppelComments = await getWithCustomLoader(context, "doppelCommentsForComment", dbComment._id, async (commentIds: string[]) => {
+          const doppelComments = await DoppelComments.find({commentId: {$in: commentIds}}).fetch()
+          const doppelsByComment = groupBy(doppelComments, 'commentId')
+          return commentIds.map(commentId => doppelsByComment[commentId] ?? [])
+        })
+        if (doppelComments.length < 2) {
+          // We'll make 'em for next time
+          void [createDoppelComment(dbComment, user), createDoppelComment(dbComment, user)]
+        }
+        return doppelComments
+      }
+    }
+  }
 });
+
+const createDoppelComment = async (dbComment: DbComment, user: DbUser) => {
+  const [interestingGwernComment, eightShortStudies, topUserComment, topUserPost] = await Promise.all([
+    Comments.findOne({ _id: 'hxMNNJRF6o644aNTa'}),
+    Posts.findOne({_id: 'gFMH3Cqw4XxwL69iy'}),
+    Comments.findOne({ userId: dbComment.userId }, { sort: { 'baseScore': -1 }, limit: 1, projection: { '_id': 1 } }),
+    Posts.findOne({ userId: dbComment.userId }, { sort: { 'baseScore': -1 }, limit: 1, projection: { '_id': 1 } }),
```

Comment:
Plausibly worth sticking these in custom loaders too, given how many users have multiple comments on a single post (such that you could often cut down on the number of database queries by an integer multiple).  Also you don't need `limit: 1` if you're doing a `findOne`.